### PR TITLE
chore(web): upgrade to react router 6

### PR DESCRIPTION
**Pull request by fallenoak**
This PR moves us up to the latest version of React Router, and switches routing from the `withRouter` HOC to hooks.

This PR tweaks the `TabbedBox` component a bit: instead of auto-redirecting to the first tab's path when tabs are mounted but not part of the URL path, we now simply show the first tab by default. This permits multiple `TabbedBox` components to co-exist in a page, and avoids dragging potentially unwanted URL fragments along for the ride when copying and pasting URLs.

### CODE REVIEW

File: packages/spelunker-web/src/components/core/Box/Tabbed/Tab.jsx [hunk 1]
```diff
@@ -1,10 +1,6 @@
 import React from 'react';
 
 const Tab = (props) => {
-  if (!props.active) {
-    return null;
-  }
-
   const Component = props.component;
   return <Component {...props} />;
 };
```
Comment (fallenoak) [inline packages/spelunker-web/src/components/core/Box/Tabbed/Tab.jsx:6]: I know this isn't doing much anymore, but I thought it'd be good to keep around as an abstraction, and to potentially filter out tab-specific props (like `path` and `label`) in the future.

Comment (timkurvers) [reply packages/spelunker-web/src/components/core/Box/Tabbed/Tab.jsx:6]: 👍

**Review by timkurvers — APPROVED** [review]
Looks good! 🥳 

I suppose one side effect of this would be that the first tab is no longer guaranteed to have a fixed canonical URL:

1) Visit an item, say: `/items/100`, with first tab 'Dropped by'
2) Database entries are changed, we roll out a new tab or flip around the order
3) Now that item's first tab could be something else

Not sure if that's an issue. +1 on the multiple tabbed boxes.

**Comment by fallenoak** [comment]
> Looks good! 🥳
> 
> I suppose one side effect of this would be that the first tab is no longer guaranteed to have a fixed canonical URL:
> 
>     1. Visit an item, say: `/items/100`, with first tab 'Dropped by'
> 
>     2. Database entries are changed, we roll out a new tab or change the tab order
> 
>     3. Now that item's first tab could be something else
> 
> 
> Not sure if that's an issue. +1 on the multiple tabbed boxes.

The first tab does still have a route, and a proper URL: to get it, you just need to eg. right click and choose copy URL from the tab, or click on the tab and copy the URL from the URL bar. I removed the auto-navigate to keep first tabs on multiple `TabbedBox`es from fighting over navigate.

We could def revisit if it seems like it's an issue. We could potentially sneak in a share link thingy in each tab if we felt like that should be addressed!

**Pull request merged**